pydantic/main.py

```diff
@@ -403,6 +404,8 @@ def model_rebuild(
             types_namespace = cls.__pydantic_parent_namespace__
 
             types_namespace = _model_construction.get_model_types_namespace(cls, types_namespace)
+            if _localns:
+                types_namespace = {**types_namespace, **_localns}
```

Comment:
I'd prefer this to be _localns only, types_namespace includes locals() or something and is not what I'd expect.
It would still be possible to have the `{**types_namespace, **_localns}` behaviour by passing 
`{**types_namespace, **_localns}` as _localns.

Reply:
okay, that makes sense, so this would be more of a hard-override